index/repo.go

```diff
@@ -0,0 +1,50 @@
+package index
+
+import (
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+// Repo is the central index repository object that manages full indices and
+// manifests.
+type Repo struct {
+	FullIndexRepo
+	ManifestRepo
+}
+
+type Stat struct {
+	Exists bool
+	Size   uint64
+}
+
+type FullIndexRepo interface {
+	// GetFullIndex returns the full index for the specified shard.
+	GetFullIndex(key shard.Key) (FullIndex, error)
+
+	// AddFullIndex persists a full index for a shard.
+	AddFullIndex(key shard.Key, index FullIndex) error
+
+	// DropFullIndex drops the full index for the specified shard. If the error
+	// is nil, it returns whether an index was effectively dropped.
+	DropFullIndex(key shard.Key) (dropped bool, err error)
```

Comment:
Question: in which scenario is dropped false?

Reply:
I intended for this to be similar to the interface of a `DELETE` command in most databases, which tells you how many rows were affected. However, since there can only be one full index, it's a binary and not a number.

In terms of logical coherence, there shouldn't be any case where `DropFullIndex` returns false, as that would indicate that an index was lost. But it may happen if the user erased an index manually from disk! For those cases, we have planned a `ReconcileIndices()` method: https://github.com/filecoin-project/dagstore/blob/master/docs/design.md#persistence-and-reconciliation